test/test_modules.py

```diff
@@ -382,12 +382,13 @@ def _can_be_noncontiguous(obj):
                     grad_output = default_output.clone().detach_().normal_()
                     default_output.backward(grad_output, retain_graph=True)
                 else:
-                    grad_output = tuple(self._traverse_obj(o, lambda o: o.clone().detach_().normal_())
+                    grad_output = tuple(self._traverse_obj(o, lambda o: o.clone().detach_().normal_() if o.requires_grad else None)
                                         for o in default_output)
                     flattened_default_output, _ = torch.utils._pytree.tree_flatten(default_output)
                     flattened_grad_output, _ = torch.utils._pytree.tree_flatten(grad_output)
                     for o, g_o in zip(flattened_default_output, flattened_grad_output):
-                        o.backward(g_o, retain_graph=True)
+                        if (o.requires_grad):
```

Comment:
For outputs like `indices` when `return_indices=True` for MaxPooling, the indices returned do not require grad so we should skip this path